### ISSUE TITLE
Double-check equation for m(t)

### ISSUE THREAD

**Issue by npow** [peterkim95/attentive-reader#1]
Is this line correct: https://github.com/peterkim95/attentive-reader/blob/master/main.py#L107
If I understand your implementation correctly, you are taking the hadamard product. And I don't see the tanh anywhere (not sure if that makes a big difference though).

```
M = d_output * tf.expand_dims(tf.matmul(q_output, W_bilinear), axis=1) # [batch, h] -> [batch, 1, h]
```

Should it be this:
```
M = d_output + tf.expand_dims(tf.matmul(q_output, W_bilinear), axis=1) # [batch, h] -> [batch, 1, h]
M = tf.nn.tanh(M)
```

![image](https://user-images.githubusercontent.com/206201/50347861-fe06be00-052e-11e9-95ba-f72c4aa27414.png)

Can you also share the results you were getting?

**Comment by tspeterkim** [comment]
Hey Nissan,

No I'm not implementing a Hadamard product. Instead I am using a bilinear term. This was how it was intended in the original paper linked [here](https://arxiv.org/pdf/1606.02858.pdf). Check out section **3.2 Attention**. The tanh step is actually what DeepMind's Attentive reader did, but in this paper they decided against it (See section **3.2 Differences from (Hermann et al., 2015)** in the same paper).

Also, the final accuracy I got for the CNN dataset was 71%. I have not actually ran it for Daily Mail.

**Comment by npow** [comment]
Thanks for clarifying! I wasn't aware of the paper from Chris Manning's group.
